modules/src/main/java/eu/itesla_project/modules/constraints/ConstraintsModifier.java

```diff
@@ -135,10 +130,10 @@ else if ( violatedBranch instanceof TwoWindingsTransformer )
                 if ( applyToBaseCase && !StateManager.INITIAL_STATE_ID.equals(stateId) ) { // change the limit also to basecase
                     String initialStateId = StateManager.INITIAL_STATE_ID;
                     network.getStateManager().setWorkingState(initialStateId);
-                    if ( violatedBranch instanceof Line )
-                        branch = network.getLine(violatedBranch.getId());
-                    else if ( violatedBranch instanceof TwoWindingsTransformer )
-                        branch = network.getTwoWindingsTransformer(violatedBranch.getId());
+                    branch = network.getLine(violation.getSubjectId());
+                    if ( branch == null ) {
+                        branch = network.getTwoWindingsTransformer(violation.getSubjectId());
+                    }
```

Comment:
Branch has already been set outside the "if" block and these lines seem to be useless?